control/control_loop.go

```diff
@@ -289,14 +289,30 @@ func (l *Loop) startBenchmark(loops int) error {
 
 // Stop stops then loop.
 func (l *Loop) Stop() {
-	if l.running {
-		l.running = false
-		l.logger.Debug("closing loop")
-		l.ct.ticker.Stop()
-		close(l.ct.stop)
-		l.cancel()
-		l.activeBackgroundWorkers.Wait()
-	}
+	l.running = false
+	l.logger.Debug("closing loop")
+	l.ct.ticker.Stop()
+	close(l.ct.stop)
+	l.cancel()
+	l.activeBackgroundWorkers.Wait()
+}
+
+// SetStatus sets l.running.
+func (l *Loop) SetStatus(running bool) {
+	l.running = running
+}
+
+func (l *Loop) Pause() {
+	l.running = false
+}
+
+func (l *Loop) Resume() {
+	l.running = true
+}
+
+// GetStatus returns the value of l.running.
+func (l *Loop) Running() bool {
+	return l.running
```

Comment:
do we need a lock for running? since we could have multiple threads accessing & modifying the field at the same time

Reply:
we weren't before and it was being read and set in multiple places? although not in as meaningful of a way so maybe?

Reply:
before this pr we only had running in Start and Stop as far as I can tell, so it was only being used by a single thread at any given time